org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semantictokens/SemanticTokensService.java

```diff
@@ -105,20 +135,20 @@ public SemanticTokens semanticTokensFull(final Document document, final XtextRes
 				Integer positionTokenType = getTokenType(lightweightPosition.getIds());
 				Integer positionTokenModifiers = getTokenModifiers(lightweightPosition.getIds());
 
-				if (positionTokenType != 0 || positionTokenModifiers != 0) {
-					int deltaOffset = lightweightPosition.getOffset() - lastOffset;
-					reader.skip(deltaOffset);
-					Position position = reader.getPosition();
-					int deltaLine = position.getLine() - lastLine;
-					data.add(deltaLine); // delta line
-					lastLine = position.getLine();
-					// delta start relative to previous token if on the same line or to 0
-					data.add(deltaLine == 0 ? deltaOffset : position.getCharacter());
-					lastOffset = lightweightPosition.getOffset();
-					data.add(lightweightPosition.getLength()); // length
-					data.add(positionTokenType); // token type
-					data.add(positionTokenModifiers); // token modifiers
-				}
+				int deltaOffset = lightweightPosition.getOffset() - lastOffset;
```

Comment:
After this change, I wonder how you'd indicate a position with an ID that is specifically not compatible to the LSP token types? E.g. another UI might support more token types.

I think that's why `0` was used as a return type for an ID that's semantic but not supported by the LSP protocol.

Reply:
the problem is, that 0 is a valid tokenTypeIndex in LSP. I now changed it to return -1 if no valid TokenType is found. So the TokenType with index 0 can be used and still we can detect unsupported types.